### PR TITLE
Fix bugs in _store_from_cache and repository.erase

### PR DESCRIPTION
**Pull request by greschd**
The `_store_from_cache` needed to erase the content of its sandbox folder before copying over the content of the cache source. Currently, the existing contents are mixed with with the contents
to be copied in.
In fixing this, we discovered an issue in repository.erase, where an unstored node would try erasing the (inexistent) folder in the permanent repository, instead of the SandboxFolder.

Fixed in collaboration with @sphuber, so this should be reviewed by someone else.

### CODE REVIEW

**Comment by greschd** [comment]
I guess we should add some regression tests for this...

**Comment by greschd** [comment]
@sphuber do you also want a regression test for the `Repository.erase` part?

**Comment by sphuber** [comment]
> @sphuber do you also want a regression test for the `Repository.erase` part?

Actually, a very simple one should not be too much work and probably give a lot of mileage. Just one where you store some files and call erase on unstored node and one where you call erase after storing (passing `force=True`). That should already cover quite a bit

**Comment by greschd** [comment]
@ltalirz in the `test_store_from_cache` (which is a new-style pytest test, yay) I now have to manually call `reset_db`, because some other part of the test suite (probably tests not using the fixtures) trashes the `User`.
Maybe it would make sense to also have a fixture which resets _before_ the tests, not only after? It seems to be a valid pattern that you need the DB to be "pristine" for a certain test.

**Comment by greschd** [comment]
Docker check failing seems unrelated to this PR.

**Comment by greschd** [comment]
For the repository tests, I added 3 cases:
- erase unstored node
- erase stored node with force=True
- erase stored node (no force flag), check that it raises

File: aiida/orm/nodes/node.py [hunk 1]
```diff
@@ -1090,6 +1090,7 @@ def _store_from_cache(self, cache_node, with_transaction):
             if key != Sealable.SEALED_KEY:
                 self.set_attribute(key, value)
 
+        self._repository.erase()
```
[the remark this answers is not in the document]
Comment (greschd) [reply aiida/orm/nodes/node.py]: Done! I'm now wondering why the `put_object_from_tree` didn't complain about the existing objects, or at least just cleanly overwrite the files (which should be the same after all).

Maybe you want to have a look at this?

**Comment by greschd** [comment]
Changed it to implicitly rely on #3783. Should start working again once that is merged and this branch is updated.

File: aiida/orm/nodes/node.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (greschd) [reply aiida/orm/nodes/node.py]: Done

**Pull request merged**